identity/README.md

```diff
@@ -0,0 +1,279 @@
+               '------------'
+```
+
+## RPC Interface
+
+**Identity Service**: The CSI-driver **MUST** implement this, the CO-plugin
+uses the Identity Service to inspect the capabilities of the CSI-driver.
+
+```protobuf
+syntax = "proto3";
+package identity;
+
+import "google/protobuf/wrappers.proto"; // for BoolValue in ProbeResponse
+
+option go_package = ".;identity";
+
+// Identity is the service that CSI-drivers MUST implement in order to be
+// compatible with CSI-Addons specification. It provides the basic details that
+// are needed by the CSI-Addons CO plugin to communicate with the CSI-driver
+// and detects its supported features.
+service Identidy {
+  // GetIdentity returns basic information about the side-car and CSI-driver.
+  rpc GetIdentity(GetIdentityRequest)
+  returns (GetIdentityResponse) {}
+
+  // GetCapabilities returns the capabilities that the CSI-driver supports.
+  rpc GetCapabilities(GetCapabilitiesRequest)
+  returns (GetCapabilitiesResponse) {}
+
+  // Probe is called by the CO plugin to validate that the CSI-Addons Node is
+  // still healthy.
+  rpc Probe(ProbeRequest)
+  returns (ProbeResponse) {}
+}
+```
+
+### `GetIdentity`
+
+The CSI-driver needs to provide its driver-name so that the CSI-Addons CO
+plugin can invoke RPCs for volumes that use this driver. There is an option for
+the CSI-driver to provide details about its version and a key/value manifest.
+
+```protobuf
+// GetIdentityRequest is sent by the CSI-Addons CO plugin to obtain the
+// drivername, version and optional details from the CSI-driver.
+message GetIdentityRequest {
+  // Intentionally empty.
+}
+
+// GetIdentityResponse is returned by the CSI-driver as a response to a
+// GetIdentityRequest.
+message GetIdentityResponse {
+  // The name MUST follow domain name notation format
+  // (https://tools.ietf.org/html/rfc1035#section-2.3.1). It SHOULD include
+  // the CSI-drivers's host company name and the CSI-driver name, to minimize
+  // the possibility of collisions. It MUST be 63 characters or less, beginning
+  // and ending with an alphanumeric character ([a-z0-9A-Z]) with dashes (-),
+  // dots (.), and alphanumerics between. This field is REQUIRED.
+  string name = 1;
+  // This field is REQUIRED. Value of this field is opaque to the CO.
+  string vendor_version = 2;
+  // This field is OPTIONAL. Values are opaque to the CO.
+  map<string, string> manifest = 3;
+}
+```
+
+#### GetIdentity Errors
+
+If the CSI-driver is unable to complete the GetIdentity call successfully, it
+MUST return a non-ok gRPC code in the gRPC status.
+
+### `GetCapabilities`
+
+The CSI-Addons plugin for the CO needs to detect the supported features for
+each CSI-driver. Features are described as capabilities, each with additional
+information on how the CSI-Addons CO plugin may invoke the operation.
+
+There might be operations that need to run on the CSI Controller (provisioner)
+for storage system access. Other operations may require running on a volume
```

Comment:
ooc, what kind of provisioner operations we can expect? any specific things in mind, if yes, it is also better to mention it?

Reply:
ReclaimSpace is the the 1st we're looking into (calling `rbd sparsify` for Ceph). Rotating encryption passphrases could be an other. Maybe some form of tuning, or setting options on a volume is possible by some SP too. Not sure it is helpful to mention here at this point in time.